### PR TITLE
Add table aws_organizations_organizational_unit and aws_organizations_root. closes #1674

### LINKED ISSUE #1674: Add table aws_organizations_organizational_unit
In AWS Organizations, the accounts can be arranged in Organizationnal units: https://docs.aws.amazon.com/organizations/latest/userguide/orgs_getting-started_concepts.html

This ticket suggests the creation of a table to enumerate these OUs.

API call to list OUs: https://docs.aws.amazon.com/organizations/latest/APIReference/API_ListOrganizationalUnitsForParent.html

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_organizations_organizational_unit where parent_id = 'r-wxnb'
+--------------------------+------------------+----------------------------------------------------------------------+-----------+--------------------------+--------------------------------------------------------------------------+-----------+--------+--------------+->
| name                     | id               | arn                                                                  | parent_id | title                    | akas                                                                     | partition | region | account_id   | >
+--------------------------+------------------+----------------------------------------------------------------------+-----------+--------------------------+--------------------------------------------------------------------------+-----------+--------+--------------+->
| Jamal                    | ou-wxnb-05v0c1hg | arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-05v0c1hg | r-wxnb    | Jamal                    | ["arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-05v0c1hg"] | aws       | global | 210987654321 | >
| Turbot SaaS Dev          | ou-wxnb-m8l8tpaq | arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-m8l8tpaq | r-wxnb    | Turbot SaaS Dev          | ["arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-m8l8tpaq"] | aws       | global | 210987654321 | >
| Turbot SaaS Staging      | ou-wxnb-wofu2g1q | arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-wofu2g1q | r-wxnb    | Turbot SaaS Staging      | ["arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-wofu2g1q"] | aws       | global | 210987654321 | >
| Devgov                   | ou-wxnb-bijhomqq | arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-bijhomqq | r-wxnb    | Devgov                   | ["arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-bijhomqq"] | aws       | global | 210987654321 | >
| www                      | ou-wxnb-vch85tma | arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-vch85tma | r-wxnb    | www                      | ["arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-vch85tma"] | aws       | global | 210987654321 | >
| Rex                      | ou-wxnb-52uc2fbu | arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-52uc2fbu | r-wxnb    | Rex                      | ["arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-52uc2fbu"] | aws       | global | 210987654321 | >
| account creation testing | ou-wxnb-feyv7riq | arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-feyv7riq | r-wxnb    | account creation testing | ["arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-feyv7riq"] | aws       | global | 210987654321 | >
| Silverwater              | ou-wxnb-v66c90u3 | arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-v66c90u3 | r-wxnb    | Silverwater              | ["arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-v66c90u3"] | aws       | global | 210987654321 | >
| Personal Sandbox         | ou-wxnb-joldicj7 | arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-joldicj7 | r-wxnb    | Personal Sandbox         | ["arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-joldicj7"] | aws       | global | 210987654321 | >
| Turbot SaaS Prod         | ou-wxnb-5rbvg65u | arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-5rbvg65u | r-wxnb    | Turbot SaaS Prod         | ["arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-5rbvg65u"] | aws       | global | 210987654321 | >
+--------------------------+------------------+----------------------------------------------------------------------+-----------+--------------------------+--------------------------------------------------------------------------+-----------+--------+--------------+-


> select * from aws_organizations_organizational_unit where id = 'ou-wxnb-05v0c1hg'
+-------+------------------+----------------------------------------------------------------------+------------------+-------+--------------------------------------------------------------------------+-----------+--------+--------------+---------------------------+
| name  | id               | arn                                                                  | parent_id        | title | akas                                                                     | partition | region | account_id   | _ctx                      |
+-------+------------------+----------------------------------------------------------------------+------------------+-------+--------------------------------------------------------------------------+-----------+--------+--------------+---------------------------+
| Jamal | ou-wxnb-05v0c1hg | arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-05v0c1hg | ou-wxnb-05v0c1hg | Jamal | ["arn:aws:organizations::123456789012:ou/o-c3a5y4wd52/ou-wxnb-05v0c1hg"] | aws       | global | 210987654321 | {"connection_name":"aws"} |
+-------+------------------+----------------------------------------------------------------------+------------------+-------+--------------------------------------------------------------------------+-----------+--------+--------------+---------------------------+

```
</details>

### CODE REVIEW

**Comment by ParthaI** [comment]
Hi @ajoga, We have added a new table `aws_organizations_organizational_unit` in this PR to address the [issue](https://github.com/turbot/steampipe-plugin-aws/issues/1674). It would be very appreciated if you could give it a try and share your experience with us. Thanks!

**Comment by misraved** [comment]
@ParthaI could you please add a few more example queries in the table docs? Maybe you need to join it with other tables to reflect its usage.

**Comment by ajoga** [comment]
Hi @ParthaI, thank you for the swift addition. Unfortunately I do not have the rights to execute that particular API call to test this case :-(
This is very likely due to my circumstances rather than your implementation, so I can't give you feedback on this.

**Comment by ajoga** [comment]
Hi again @ParthaI, I'll elaborate on my limits as maybe this will impact #1675 

I can call [ListChildren](https://docs.aws.amazon.com/organizations/latest/APIReference/API_ListChildren.html) that can yield either OUs or AWS accounts. 

As I suggested the addition of that table so Steampipe can then link accounts back to OUs (which is, iirc, an information you can only have from the AWS organization account), maybe you would have an interest to use ListChildren rather than [ListAccountsForParent](https://docs.aws.amazon.com/organizations/latest/APIReference/API_ListAccountsForParent.html)? 

I'm thinking out loud here and obviously biased toward using ListChildren rather than ListAccountsForParent, so feel free to not go this way.

File: docs/tables/aws_organizations_organizational_unit.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Table: aws_organizations_organizational_unit
+
+A container for accounts within a root. An OU also can contain other OUs, enabling you to create a hierarchy that resembles an upside-down tree, with a root at the top and branches of OUs that reach down, ending in accounts that are the leaves of the tree. When you attach a policy to one of the nodes in the hierarchy, it flows down and affects all the branches (OUs) and leaves (accounts) beneath it. An OU can have exactly one parent, and currently each account can be a member of exactly one OU.
+
+**Note**: The `parent_id` is the required to make the API call. It is the unique identifier (ID) of the root or OU whose child OUs you want to list.
```
Comment (ajoga) [inline docs/tables/aws_organizations_organizational_unit.md:5]: Your second example in this ticket *does not* specify `parent_id`, how come it worked?

**Comment by ParthaI** [comment]
> Hi again @ParthaI, I'll elaborate on my limits as maybe this will impact #1675
> 
> I can call [ListChildren](https://docs.aws.amazon.com/organizations/latest/APIReference/API_ListChildren.html) that can yield either OUs or AWS accounts.
> 
> As I suggested the addition of that table so Steampipe can then link accounts back to OUs (which is, iirc, an information you can only have from the AWS organization account), maybe you would have an interest to use ListChildren rather than [ListAccountsForParent](https://docs.aws.amazon.com/organizations/latest/APIReference/API_ListAccountsForParent.html)?
> 
> I'm thinking out loud here and obviously biased toward using ListChildren rather than ListAccountsForParent, so feel free to not go this way.

@ajoga Thanks for the suggestion will play around those APIs and will let you know. Thanks again.

File: docs/tables/aws_organizations_organizational_unit.md (hunk 1 above)
Comment (ParthaI) [reply docs/tables/aws_organizations_organizational_unit.md:5]: > Your second example in this ticket _does not_ specify `parent_id`, how come it worked?

In the table definition we have two config one is [List Config](https://github.com/turbot/steampipe-plugin-aws/pull/1677/files#diff-a5533f317b4a9cd8f0738573bf490499b95488f94416241aad2bcd8729fcd650R27) and another one is [Get Config](https://github.com/turbot/steampipe-plugin-aws/pull/1677/files#diff-a5533f317b4a9cd8f0738573bf490499b95488f94416241aad2bcd8729fcd650R20).
In List config we generally make list API call and in Get config we make the get API call.

Here we are using [ListAccountsForParent](https://docs.aws.amazon.com/organizations/latest/APIReference/API_ListOrganizationalUnitsForParent.html) API in the `List config`, for making the `ListAccountsForParent` API call we must need to pass the `ParentId` in param of this API call, so `parent_id` is required to make list API call.

In Get config we are using the [DescribeOrganizationalUnit](https://docs.aws.amazon.com/organizations/latest/APIReference/API_DescribeOrganizationalUnit.html) API call, for making this API call we must have to pass the `OrganizationalUnitId`, we do not need `ParentId` for this API call.

The Get API will be called if we are providing `id` as query parameter, so in second example we need to pass the `parent_id`.

Comment (ajoga) [reply docs/tables/aws_organizations_organizational_unit.md:5]: Got it for the logic underneath ; may I suggest a working like this? 

```suggestion
You **_must_** specify a single `parent_id` or `id` in a where or join clause in order to use this table.
```
I found similar wording/format for [aws_route53_record](https://hub.steampipe.io/plugins/turbot/aws/tables/aws_route53_record), [aws_iam_access_advisor](https://hub.steampipe.io/plugins/turbot/aws/tables/aws_iam_access_advisor), [aws_cloudtrail_trail_event](https://hub.steampipe.io/plugins/turbot/aws/tables/aws_cloudtrail_trail_event) and others

Also from a user perspective I find it odd that we can't have the listing without a parent_id, but looking at API reference it looks tricky to get the root id reliably...

Comment (ParthaI) [reply docs/tables/aws_organizations_organizational_unit.md:5]: Yeah @ajoga, this looks better, will push the changes. Thanks!

File: aws/table_aws_organizations_organizational_unit.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+		ParentId:   aws.String(parentId),
+		MaxResults: &maxItems,
+	}
+
+	paginator := organizations.NewListOrganizationalUnitsForParentPaginator(svc, params, func(o *organizations.ListOrganizationalUnitsForParentPaginatorOptions) {
+		o.Limit = maxItems
+		o.StopOnDuplicateToken = true
+	})
+
+	for paginator.HasMorePages() {
```
Comment (madhushreeray30) [inline aws/table_aws_organizations_organizational_unit.go:146]:
```suggestion
	for paginator.HasMorePages() {
	// apply rate limiting
```

File: aws/table_aws_organizations_root.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	params := &organizations.ListRootsInput{
+		MaxResults: &maxItems,
+	}
+
+	paginator := organizations.NewListRootsPaginator(svc, params, func(o *organizations.ListRootsPaginatorOptions) {
+		o.Limit = maxItems
+		o.StopOnDuplicateToken = true
+	})
+
+	for paginator.HasMorePages() {
```
Comment (madhushreeray30) [inline aws/table_aws_organizations_root.go:96]:
```suggestion
	for paginator.HasMorePages() {
	// apply rate limiting
```

File: docs/tables/aws_organizations_organizational_unit.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+select
+  name,
+  id,
+  parent_id,
+  path
+from
+  aws_organizations_organizational_unit
+where
+  (length(path) - length(replace(path, '.', ''))) = 2;
+
```
Comment (madhushreeray30) [inline docs/tables/aws_organizations_organizational_unit.md:90]:
```suggestion
```

File: docs/tables/aws_organizations_organizational_unit.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    from
+      aws_organizations_organizational_unit
+    where
+      name = 'Punisher'
+  );
+```
+
+### Select organizational units with a path that matches a specific pattern
+
+```sql
```
Comment (madhushreeray30) [inline docs/tables/aws_organizations_organizational_unit.md:165]: Please add queries for both `postgres` and `sqlite`

**Review by madhushreeray30 — APPROVED** [review]

**Pull request merged**